### PR TITLE
Further AppKit additions

### PR DESCRIPTION
**Pull request by tonyarnold**
This PR further fleshes out Bond's AppKit support by adding drastically improved support for `NSButton`. 

If the changes in this PR are appropriate, I'd be happy to walk through and flesh out the other classes, as well as adding anything that I think is important. 

Should I be using `DynamicSubject` this much? I don't want to over complicate your library.

### CODE REVIEW

**Comment by srdanrasic** [comment]
Is AppKit KVO compatible? :)

What's the reason behind having all these as DynamicSubjects instead of Bonds?

**Comment by tonyarnold** [comment]
Most all of AppKit is, yes (unlike UIKit). I made these DynamicSubjects because Bonds are set-only, aren't they?

**Comment by tonyarnold** [comment]
Following our conversation on Gitter, I've changed all of the DynamicSubjects into Bonds, and exposed willChangeSelection/didChangeSelection on `NSTableView`. 

I get it now 🙄 😀

File: Sources/AppKit/NSTableView.swift [hunk 1]
```diff
@@ -18,6 +18,22 @@ public extension ReactiveExtensions where Base: NSTableView {
   public var dataSource: ProtocolProxy {
     return base.protocolProxy(for: NSTableViewDataSource.self, setter: NSSelectorFromString("setDataSource:"))
   }
+
+  public var selectionWillChange: SafeSignal<Void> {
+    return NotificationCenter.default.reactive.notification(name: .NSTableViewSelectionWillChange).eraseType()
+  }
+
+  public var selectionDidChange: SafeSignal<Void> {
+    return NotificationCenter.default.reactive.notification(name: .NSTableViewSelectionDidChange).eraseType()
```
Comment (srdanrasic) [inline Sources/AppKit/NSTableView.swift]: We should probably set `object: self` here and above :)

Comment (tonyarnold) [inline Sources/AppKit/NSTableView.swift]: Oh, god 🤦‍♂️ I've been using this API (NSNotificationCenter) for over a decade, and I still get it wrong…

**Comment by srdanrasic** [comment]
This looks great - merging :)

**Pull request merged**